Library/Homebrew/formula.rb

```diff
@@ -775,6 +775,8 @@ def to_hash
           "used_options" => tab.used_options.as_flags,
           "built_as_bottle" => tab.built_bottle,
           "poured_from_bottle" => tab.poured_from_bottle
+          if ENV["HOMEBREW_CXX"]: "CXX_Compiler" => ENV["HOMEBREW_CXX"]
```

Comment:
I'd be fine with including it. We haven't stored this before simply because it hasn't been requested in the past, and because users typically have access to the build logs in `~/Library/Logs/Homebrew` that makes it possible to determine what compiler was used.

What is your usecase, out of curiosity?